qcodes/instrument_drivers/AlazarTech/ats_api.py

```diff
@@ -0,0 +1,502 @@
+"""
+This module provides a class that encapsulates Alazar ATS API, 
+:class:`AlazarATSAPI`. The class is used to expose Alazar API functions
+of its C library in a python-friendly way.
+"""
+
+
+from typing import Dict, Tuple, Union, Any
+import ctypes
+from ctypes import POINTER
+
+from .dll_wrapper import WrappedDll, Signature
+from .constants import BOARD_NAMES, REGISTER_ACCESS_PASSWORD, ReturnCode
+
+# `_BaseParameter` is needed because users may pass instrument parameters
+# that originate from `Instrument.parameters` dictionary which is typed
+# with `_BaseParameter`, not `Parameter`.
+from qcodes.instrument.parameter import _BaseParameter as Parameter
+
+# Define aliases for ctypes that match Alazar's notation.
+U8 = ctypes.c_uint8
+U32 = ctypes.c_uint32
+HANDLE = ctypes.c_void_p
+
+POINTER_U8 = Any
+POINTER_U32 = Any
+
```

Comment:
Could we give these slightly different names so that it's clear that. U8 and U32 are ctypes types while (POINTER_U8 and POINTER_U32) are just typing aliases with no implementation. I.e they are only used for typing, in the code you use POINTER(U8)

Reply:
I assume this is needed since you cant use POINTER(U8) in the type signature

Reply:
exactly, i wasted quite some time trying to figure out the way to do it properly, and failed so far. googling didn't help either, seems that people never do this.

Shall I rename them to `POINTER_c_uint32`/`POINTER_c_uint8` so that they look more like `ctypes` types?